libs/wingc/src/type_check/symbol_env.rs

```diff
@@ -132,6 +147,23 @@ impl SymbolEnv {
 		self.parent.is_none()
 	}
 
+	/// Checks if this environment is a parent of another environment.
+	pub fn is_parent_of(&self, other: &SymbolEnv) -> bool {
+		let Some(parent) = other.parent else {
+			return false;
+		};
+
+		if self.is_same(&parent) {
+			return true;
+		}
+
+		self.is_parent_of(&parent)
+	}
+
+	pub fn is_same(&self, other: &SymbolEnv) -> bool {
+		std::ptr::eq(&*other, &*self)
```

Comment:
Since we're not using a `SymbolEnvRef` here jsut a `&SymbolEnv` I suspect this will work too:
```suggestion
		std::ptr::eq(other, self)
```

Reply:
Trying...

Reply:
Works.